addons/account/models/sequence_mixin.py

```diff
@@ -58,11 +62,20 @@ def _sequence_matches_date(self):
         if not sequence or not date:
             return True
 
+        fyear_start = fyear_end = False
+        if 'company_id' in self:
+            company = self.company_id
+            fyear_start, fyear_end = date_utils.get_fiscal_year(date, day=company.fiscalyear_last_day, month=int(company.fiscalyear_last_month))
         format_values = self._get_sequence_format_param(sequence)[1]
         year_match = (
             not format_values["year"]
-            or format_values["year"] == date.year % 10 ** len(str(format_values["year"]))
+            or self._year_match(format_values["year"], date)
         )
+        if format_values["fyear_start"] and format_values["fyear_end"]:
+            year_match = (
+                self._year_match(format_values["fyear_start"], fyear_start or date)
+                and self._year_match(format_values["fyear_end"], fyear_end or date)
+            )
```

Comment:
I still think the logics of this is confused.
`year_match` gets overwritten... `fyear_start` and `fyear_end` only get populated `if 'company_id' in self` and used only on the second `if`...

```py
        if format_values["year"]:
            year_match = self._year_match(format_values["year"], date)
        elif 'company_id' in self and format_values["fyear_start"] and format_values["fyear_end"]:
            company = self.company_id
            fyear_start, fyear_end = date_utils.get_fiscal_year(date, day=company.fiscalyear_last_day, month=int(company.fiscalyear_last_month))
            year_match = (
                self._year_match(format_values["fyear_start"], fyear_start or date)
                and self._year_match(format_values["fyear_end"], fyear_end or date)
            )
        else:
            year_match = True
```

Reply:
can't do it like this because if we get `fyear_start` and `fyear_end` then force overwrite 'year_match'

Reply:
That's not the point, we can adapt it...
```py
        # this stays true if not format_values["year"]
        year_match = True 
        if format_values["year"]:
            year_match = self._year_match(format_values["year"], date)
        if 'company_id' in self and format_values["fyear_start"] and format_values["fyear_end"]:
            company = self.company_id
            fyear_start, fyear_end = date_utils.get_fiscal_year(date, day=company.fiscalyear_last_day, month=int(company.fiscalyear_last_month))
            year_match = (
                self._year_match(format_values["fyear_start"], fyear_start or date)
                and self._year_match(format_values["fyear_end"], fyear_end or date)
            )
```

... otherwise if this is true then we have a traceback because `fyear_start` and `fyear_end` are not defined.
```py
    'company_id' is not in self and format_values["fyear_start"] and format_values["fyear_end"]`
```

Reply:
@lordkrandel it's done